contracts/examples/LiquidityExamples.sol

```diff
@@ -0,0 +1,234 @@
+
+    /// @notice Calls the mint function defined in perphery, mints the same amount of each token. For this example we are providing 1000 DAI and 1000 USDC in liquidity
+    /// @return tokenId The id of the newly minted ERC721
+    /// @return liquidity The amount of liquidity for the position
+    /// @return amount0 The amount of token0
+    /// @return amount1 The amount of token1
+    function mintNewPosition()
+        external
+        returns (
+            uint256 tokenId,
+            uint128 liquidity,
+            uint256 amount0,
+            uint256 amount1
+        )
+    {
+        // For this example, we will provide equal amounts of liquidity in both assets.
+        // Providing liquidity in both assets means liquidity will be earning fees and is considered in-range.
+        uint256 amount0ToMint = 1000;
+        uint256 amount1ToMint = 1000;
+
+        // Approve the position manager
+        TransferHelper.safeApprove(DAI, address(nonfungiblePositionManager), amount0ToMint);
+        TransferHelper.safeApprove(USDC, address(nonfungiblePositionManager), amount1ToMint);
+
+        INonfungiblePositionManager.MintParams memory params =
+            INonfungiblePositionManager.MintParams({
+                token0: DAI,
+                token1: USDC,
+                fee: poolFee,
+                tickLower: TickMath.MIN_TICK,
+                tickUpper: TickMath.MAX_TICK,
+                amount0Desired: amount0ToMint,
+                amount1Desired: amount1ToMint,
+                amount0Min: 0,
+                amount1Min: 0,
+                recipient: address(this),
+                deadline: block.timestamp
+            });
+
+        // Note that the pool defined by DAI/USDC and fee tier 0.3% must already be created and initialized in order to mint
+        (tokenId, liquidity, amount0, amount1) = nonfungiblePositionManager.mint(params);
+
+        // Create a deposit
+        _createDeposit(msg.sender, tokenId);
+
+        // Remove allowance and refund in both assets.
+        if (amount0 < amount0ToMint) {
+            TransferHelper.safeApprove(DAI, address(nonfungiblePositionManager), 0);
+            uint256 refund0 = amount0ToMint - amount0;
+            TransferHelper.safeTransfer(DAI, msg.sender, refund0);
+        }
+
+        if (amount1 < amount1ToMint) {
+            TransferHelper.safeApprove(USDC, address(nonfungiblePositionManager), 0);
+            uint256 refund1 = amount1ToMint - amount1;
+            TransferHelper.safeTransfer(USDC, msg.sender, refund1);
+        }
+    }
+
+    /// @notice Collects the fees associated with provided liquidity
+    /// @dev The contract must hold the erc721 token before it can collect fees
+    /// @param tokenId The id of the erc721 token
+    /// @return amount0 The amount of fees collected in token0
+    /// @return amount1 The amount of fees collected in token1
+    function collectAllFees(uint256 tokenId) external returns (uint256 amount0, uint256 amount1) {
```

Comment:
Might make more sense to collect the fees from an existing deposit, rather than create a new deposit while collecting fees? Seems a bit backwards in a way...